### PR TITLE
Bug 1383138: suggest 'rollout latest' if 'deploy --latest' returns a bad request

### PR DESCRIPTION
**Pull request by kargakis**
@mfojtik ptal

### CODE REVIEW

File: pkg/deploy/registry/instantiate/rest.go [hunk 1]
```diff
@@ -47,7 +47,7 @@ func (s *REST) New() runtime.Object {
 func (r *REST) Create(ctx kapi.Context, obj runtime.Object) (runtime.Object, error) {
 	req, ok := obj.(*deployapi.DeploymentRequest)
 	if !ok {
-		return nil, errors.NewBadRequest(fmt.Sprintf("wrong object passed for requesting a new deployment: %#v", obj))
+		return nil, errors.NewInternalError(fmt.Errorf("wrong object passed for requesting a new rollout: %#v", obj))
```
Comment (mfojtik) [inline pkg/deploy/registry/instantiate/rest.go]: can we add the suggestion here (on server side)

Comment (kargakis) [inline pkg/deploy/registry/instantiate/rest.go]: No because the request may not come from oc

**Comment by kargakis** [comment]
[test]

**Comment by mfojtik** [comment]
[merge]

**Comment by kargakis** [comment]
[test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/8571 [test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/11094 [test]

**Comment by kargakis** [comment]
https://github.com/openshift/origin/issues/11094 [merge]

**Pull request merged**